pkg/cloud/scope/cluster.go

```diff
@@ -184,6 +184,14 @@ func (s *ClusterScope) ControlPlaneLoadBalancerScheme() infrav1.ClassicELBScheme
 	return infrav1.ClassicELBSchemeInternetFacing
 }
 
+// ControlPlaneLoadBalancerName returns the Classic ELB name (user-defined, or generated from the cluster name).
+func (s *ClusterScope) ControlPlaneLoadBalancerName() (string, error) {
+	if s.ControlPlaneLoadBalancer() != nil && s.ControlPlaneLoadBalancer().Name != nil {
+		return *s.ControlPlaneLoadBalancer().Name, nil
+	}
+	return infrav1.GenerateELBName(s.Name())
```

Comment:
Do we need this and the defaulting webhook change?

Reply:
Not sure I understand your question.

The defaulting webhook change allows a user to define the Name field on create, and define it on Update _as long as the value is whatever would have been generated_.